sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -20,247 +19,424 @@
+specify any other value for `Layout`. When `Use` has any other value,
+there is no default for `Layout`, and the application must specify one
+explicitly.
+
+=== Collective matrix operations
+The following operations (load, store, multiply-and-add, fill, and
+element-wise operations) are group functions as defined in section
+4.17.3 of the core SYCL specification. As such, they must be
+encountered in convergent control flow by the work-items in the group
+that performs the group operation. The `Group` template argument must
+be `sycl::sub_group`. This extension currently supports
+only the sub-group scope, so the `Group` template parameter must be
+`sycl::sub_group`.
+
+==== Load
 ```c++
 namespace sycl::ext::oneapi::experimental::matrix {
-  template <typename Group, typename T, size_t NumRows, size_t NumCols,
-            access::address_space Space>
-  void joint_matrix_load(Group sg,
-    joint_matrix<Group, T, use::accumulator, NumRows, NumCols, layout::dynamic> &res,
-    multi_ptr<T, Space, IsDecorated> src, size_t stride, layout Layout);
-    
-  template <typename Group, typename T, size_t NumRows, size_t NumCols,
-          use Use, layout Layout, access::address_space Space>
-  void joint_matrix_load(Group sg,
-    joint_matrix<Group, T, Use, NumRows, NumCols, Layout> &res,
-    multi_ptr<T, Space, IsDecorated> src, size_t stride);
-}
-```
 
-`joint_matrix_load` loads data from memory to the 2d tiles/registers of the tensor hardware.
-We define two overloads of the load function depending on whether the memory layout was declared as part of the `joint_matrix` type or not. 
-The first overload that takes memory layout as an argument is only available for a `joint_matrix` type that used the default value `layout::dynamic`.
-The second overload without a memory layout must not be used with a `joint_matrix` type that used the default value `layout::dynamic`.
+template <typename Group, typename T,
+          size_t Rows, size_t Cols,
+          access::address_space Space, access::decorated IsDecorated>
+void joint_matrix_load(Group g,
+    joint_matrix<Group, T, use::accumulator, Rows, Cols,
+    layout::dynamic> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride, layout Layout);
 
-The base pointer `src` here determines the starting address of the matrix to be loaded from. `Layout` determines whether the data is being read in a row (`row_major`), column major (`column_major`) fashion. `stride` describes the number of elements between consecutive rows for the row major layout, or between columns for the column major layout. 
+// Only available when Layout != layout::dynamic
+template <typename Group, typename T,
+          size_t Rows, size_t Cols,
+          use Use, layout Layout,
+          access::address_space Space, access::decorated IsDecorated>
+void joint_matrix_load(Group g,
+    joint_matrix<Group, T, Use, Rows, Cols, Layout> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride);
```

Comment:
Yeah at some point raw pointers should be supported: There was an idea that it would eventually be possible to cast a raw pointer to a multi_ptr which would mean that we don't duplicate all these function implementation for the USM case. But I'm not sure what the status of this is.